pkg/mimir/tracing.go

```diff
@@ -36,3 +42,195 @@ type wrappedServerStream struct {
 func (ss wrappedServerStream) Context() context.Context {
 	return ss.ctx
 }
+
+type OpenTelemetryProviderBridge struct {
+	tracer opentracing.Tracer
+}
+
+func NewOpenTelemetryProviderBridge(tracer opentracing.Tracer) *OpenTelemetryProviderBridge {
+	return &OpenTelemetryProviderBridge{
+		tracer: tracer,
+	}
+}
+
+// Tracer creates an implementation of the Tracer interface.
+// The instrumentationName must be the name of the library providing
+// instrumentation. This name may be the same as the instrumented code
+// only if that code provides built-in instrumentation. If the
+// instrumentationName is empty, then a implementation defined default
+// name will be used instead.
+//
+// This method must be concurrency safe.
+func (p *OpenTelemetryProviderBridge) Tracer(instrumentationName string, opts ...trace.TracerOption) trace.Tracer {
+	return NewOpenTelemetricTracerBridge(p.tracer, p)
+}
+
+type OpenTelemetricTracerBridge struct {
+	tracer   opentracing.Tracer
+	provider trace.TracerProvider
+}
+
+func NewOpenTelemetricTracerBridge(tracer opentracing.Tracer, provider trace.TracerProvider) *OpenTelemetricTracerBridge {
+	return &OpenTelemetricTracerBridge{
+		tracer:   tracer,
+		provider: provider,
+	}
+}
+
+// Start creates a span and a context.Context containing the newly-created span.
+//
+// If the context.Context provided in `ctx` contains a Span then the newly-created
+// Span will be a child of that span, otherwise it will be a root span. This behavior
+// can be overridden by providing `WithNewRoot()` as a SpanOption, causing the
+// newly-created Span to be a root span even if `ctx` contains a Span.
+//
+// When creating a Span it is recommended to provide all known span attributes using
+// the `WithAttributes()` SpanOption as samplers will only have access to the
+// attributes provided when a Span is created.
+//
+// Any Span that is created MUST also be ended. This is the responsibility of the user.
+// Implementations of this API may leak memory or other resources if Spans are not ended.
+func (t *OpenTelemetricTracerBridge) Start(ctx context.Context, spanName string, opts ...trace.SpanStartOption) (context.Context, trace.Span) {
+	var mappedOptions []opentracing.StartSpanOption
+
+	// Map supported options.
+	if len(opts) > 0 {
+		mappedOptions = make([]opentracing.StartSpanOption, 0, len(opts))
+		cfg := trace.NewSpanStartConfig(opts...)
```

Comment:
I've tried but I think it's non trivial, because of the data type differences. I'm not sure it's worth to spend much time on it, at least for our use case.

Reply:
ok, sounds good. the spans seem well ordered, so if this causes a bug, we can revisit